homeworks/lesia.turua_lesiaturuta/PokemonProject/scripts/pokemon.js

```diff
@@ -309,58 +66,131 @@ const renderCards = (cardContainer, pokemonsArray) => {
     return true;
 };
 
-const cleanElement = (el) => {
-    while (el.firstChild) {
-        el.removeChild(el.firstChild);
+const renderSelect = (name) => {
+    cleanElement(searchEl);
+    const optionFirstEl = document.createElement('option');
+    optionFirstEl.value = '';
+    optionFirstEl.text = 'All types';
+    searchEl.add(optionFirstEl);
+    typesPokemons.forEach((type) => {
+        const optionEl = document.createElement('option');
+        optionEl.value = type;
+        if (type === name) {
+            optionEl.selected = true;
+        }
+        optionEl.text = type;
+        searchEl.add(optionEl);
+    });
+};
+
+const sortPokemonsUp = (arrayPokemons) => arrayPokemons.sort((a, b) => a.height - b.height);
+
+const sortPokemonsDown = (arrayPokemons) => arrayPokemons.sort((a, b) => b.height - a.height);
+
+const getPokemonTypes = (arrayPokemons) => {
+    const types = [];
+    arrayPokemons.forEach((pokemon) => pokemon.type.forEach((type) => {
+        if (!types.includes(type)) types.push(type);
+    }));
+    return types;
+};
+
+const getPokemonsByType = (arrayPokemons) => (typeValue
+    ? arrayPokemons
+        .filter((pokemon) => pokemon.type
+            .includes(typeValue))
+    : arrayPokemons);
+
+const getPokemonsByName = (arrayPokemons) => (searchValue
+    ? arrayPokemons
+        .filter((pokemon) => pokemon.name
+            .toLocaleLowerCase()
+            .includes(searchValue.toLocaleLowerCase()))
+    : arrayPokemons);
+
+const getPokemonsByWeight = (arrayPokemons) => {
+    let newPokemons = [...arrayPokemons];
+    if (checkboxBig && checkboxSmall && newPokemons.length) {
+        newPokemons = newPokemons.filter((pokemon) => pokemon.weight < 100 && pokemon.weight > 50);
```

Comment:
```suggestion
        newPokemons = newPokemons.filter((pokemon) => pokemon.weight > 100 && pokemon.weight < 50);
```
Please fix in the final version